modules/EvseManager/EvseManager.cpp

```diff
@@ -243,21 +243,28 @@ void EvseManager::ready() {
         auto sae_mode = types::iso15118_charger::SaeJ2847BidiMode::None;
 
         // Set up energy transfer modes for HLC. For now we only support either DC or AC, not both at the same time.
-        std::vector<types::iso15118_charger::EnergyTransferMode> transfer_modes;
+        std::vector<types::iso15118_charger::SupportedEnergyMode> transfer_modes;
         if (config.charge_mode == "AC") {
             types::iso15118_charger::SetupPhysicalValues setup_physical_values;
             setup_physical_values.ac_nominal_voltage = config.ac_nominal_voltage;
             r_hlc[0]->call_set_charging_parameters(setup_physical_values);
 
+            constexpr auto support_bidi = false;
+
             // FIXME: we cannot change this during run time at the moment. Refactor ISO interface to exclude transfer
             // modes from setup
             // transfer_modes.push_back(types::iso15118_charger::EnergyTransferMode::AC_single_phase_core);
-            transfer_modes.push_back(types::iso15118_charger::EnergyTransferMode::AC_three_phase_core);
+            transfer_modes.push_back({types::iso15118_charger::EnergyTransferMode::AC_three_phase_core, support_bidi});
 
         } else if (config.charge_mode == "DC") {
-            transfer_modes.push_back(types::iso15118_charger::EnergyTransferMode::DC_extended);
+            transfer_modes.push_back({types::iso15118_charger::EnergyTransferMode::DC_extended, false});
+
+            const auto caps = get_powersupply_capabilities();
+            update_powersupply_capabilities(caps);
 
-            update_powersupply_capabilities(get_powersupply_capabilities());
+            if (caps.bidirectional) {
```

Comment:
ok for now, but I'm not sure if this still works if the power supply capabilties are not yet available here. In this case they will default to safe defaults and the real values/bidi flag will come once connection to the power supply is established. Later this probably needs to go into update_powersupply_capabilities().

Reply:
Hmm, then the EvseManager should call `call_setup` from the iso15118_charger interface only if the power supply caps are available and not before.

Reply:
It may change any time, so a power supply may also sometimes be able to do bidi and some times not. Whenever that changes, we need to update the iso module. This is why we factored out e.g. set_charging_parameters from setup(). Setup() should only contain things that are fixed and available at start up, everything else should be runtime-modifiable, this includes the min/max power ratings and the bidi flags.

Reply:
I see